### PR TITLE
feat：add github toolkit plugin

### PR DESCRIPTION
**Pull request by Yeaury**
### Describe what this PR does / why we need it

 Call the official RESTAPI to implement operations on GitHub repositories

### Does this pull request fix one issue?

#98

### CODE REVIEW

**Comment by Yeaury** [comment]
Implement a few basic functions first, see if such a structure works, and add the rest

**Comment by Yeaury** [comment]
Implement some basic functionality first, see if such a structure is allowed, and then add the rest

File: community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-deploy-plugin</artifactId>
+                <version>${maven-deploy-plugin.version}</version>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml:75]: add fileName spring-ai-alibaba-plugin-github

File: community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml:64]: remove plugins

File: community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.alibaba.cloud.ai</groupId>
+        <artifactId>spring-ai-alibaba</artifactId>
+        <version>${revision}</version>
+        <relativePath>../../../pom.xml</relativePath>
+    </parent>
+    <artifactId>spring-ai-alibaba-starter-plugin-github</artifactId>
+    <name>spring-ai-alibaba-starter-plugin-math</name>
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/pom.xml:29]: update name、description to github

File: community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubAutoConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+package com.alibaba.cloud.ai.plugin.github;
+
+import com.alibaba.cloud.ai.plugin.github.service.Impl.*;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Description;
+
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubAutoConfiguration.java:26]: add @author

File: community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubAutoConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+@Configuration
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubAutoConfiguration.java:27]: remove @configuration annotation

File: community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubProperties.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.ai.plugin.github;
+
```
Comment (PolarishT) [inline community/plugins/spring-ai-alibaba-starter-plugin-github/src/main/java/com/alibaba/cloud/ai/plugin/github/GithubProperties.java:19]: add @author

**Review by PolarishT — CHANGES_REQUESTED** [review]
please follow the guideline https://github.com/alibaba/spring-ai-alibaba/blob/main/community/plugins/README.md to fix the problems marked.

**Comment by PolarishT** [comment]
please move the service implements to parent folder.

**Pull request closed without merging**